src/Razor/src/Microsoft.VisualStudio.LegacyEditor.Razor/RazorDocumentManager.cs

```diff
@@ -22,12 +21,12 @@ internal sealed class RazorDocumentManager(
     JoinableTaskContext joinableTaskContext) : IRazorDocumentManager
 {
     private readonly ProjectSnapshotManagerDispatcher _dispatcher = dispatcher;
-    private readonly JoinableTaskContext _joinableTaskContext = joinableTaskContext;
+    private readonly JoinableTaskFactory _jtf = joinableTaskContext.Factory;
     private readonly IRazorEditorFactoryService _editorFactoryService = editorFactoryService;
 
     public async Task OnTextViewOpenedAsync(ITextView textView, IEnumerable<ITextBuffer> subjectBuffers)
     {
-        _joinableTaskContext.AssertUIThread();
+        await _jtf.SwitchToMainThreadAsync();
```

Comment:
there are many other instances of using `JoinableTaskContext`, for example VisualStudioRazorParser which your PR also has a diff on has one more instance usage of that.

do we want to change that to also use `JoinableTaskFactory`?

Reply:
I ask because just yesterday I came across a UI delay on `Microsoft.VisualStudio.Editor.Razor.RazorSyntaxTreePartialParser.Parse(Microsoft.AspNetCore.Razor.Language.SourceChange)` that consumed 10+ seconds of UI thread time. Doesn't seem like a commonly hit issue though.

Reply:
Much of the JTF use in the Razor code should definitely be audited. It's possible that doing work to thread it into `VisualStudioRazorParser` could help that UI delay. However, note that `VisualStudioRazorParser` and `RazorSyntaxTreePartialParser` are only used by the legacy editor. In any case, such a change is outside of the scope of this PR. Could you file a bug to explore your suggestion?